src/logPointsManager.ts

```diff
@@ -0,0 +1,420 @@
+
+    enumerateProcesses(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User): Promise<CommandRunResult<EnumerateProcessResponse>>;
+
+    attachProcess(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User, data: AttachProcessRequest): Promise<CommandRunResult<AttachProcessResponse>>;
+}
+
+abstract class LogPointsDebuggerClientBase {
+    abstract call<ResponseType>(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User, command: string): Promise<CommandRunResult<ResponseType>>;
+
+    protected makeCallAndLogException<ResponseType>(site: WebSiteModels.Site, affinityValue: string,
+        publishCredential: WebSiteModels.User, command: string): Promise<CommandRunResult<ResponseType>> {
+        return this.call<ResponseType>(site, affinityValue, publishCredential, command)
+            .catch<CommandRunResult<ResponseType>>((err) => {
+                util.getOutputChannel().appendLine(`API call error ${err.toString()}`);
+                throw err;
+            });;
+    }
+}
+
+class KuduLogPointsDebuggerClient extends LogPointsDebuggerClientBase implements LogPointsDebuggerClient {
+    startSession(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User)
+        : Promise<CommandRunResult<StartSessionResponse>> {
+        // TODO: The actual command is TBD
+        return this.makeCallAndLogException<StartSessionResponse>(site, affinityValue, publishCredential, "node -v");
+    }
+
+    enumerateProcesses(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User): Promise<CommandRunResult<EnumerateProcessResponse>> {
+        // TODO: The actual command is TBD
+        return this.makeCallAndLogException<EnumerateProcessResponse>(site, affinityValue, publishCredential, "node -v");
+    }
+
+    attachProcess(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User): Promise<CommandRunResult<AttachProcessResponse>> {
+        // TODO: The actual command is TBD
+        return this.makeCallAndLogException<AttachProcessResponse>(site, affinityValue, publishCredential, "node -v");
+    }
+
+    call<ResponseType>(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User, command: string)
+        : Promise<CommandRunResult<ResponseType>> {
+        const siteName = util.extractSiteName(site) + (util.isSiteDeploymentSlot(site) ? '-' + util.extractDeploymentSlotName(site) : '');
+
+        let headers = {
+            "Authorization": "Basic " +
+            new Buffer(publishCredential.publishingUserName + ":" + publishCredential.publishingPassword)
+                .toString("base64")
+        };
+
+        var r = req.defaults({
+            baseUrl: "https://" + siteName + ".scm.azurewebsites.net/",
+            headers: headers,
+        });
+
+        r.cookie(`ARRAffinity=${affinityValue}`);

```

Comment:
This is the reason why we send rest calls manually. Because the KuduClient dependency does not seem to support setting cookies. If this is not the case, please let us know.